nodejs/yarn_repositories.bzl

```diff
@@ -0,0 +1,217 @@
+    yarn_script = "%s/bin/yarn.js" % yarn_path
+    yarn_entry = "bin/yarn%s" % entry_ext
+    yarn_script_relative = paths.relativize(yarn_script, "bin")
+    node_label = get_node_label(repository_ctx)
+
+    # The entry points for yarn for osx/linux and windows.
+    # Runs yarn using appropriate node entry point.
+    # Unset YARN_IGNORE_PATH before calling yarn incase it is set so that
+    # .yarnrc yarn-path is followed if set. This is for the case when calling
+    # bazel from yarn with `yarn bazel ...` and yarn follows yarn-path in
+    # .yarnrc it will set YARN_IGNORE_PATH=1 which will prevent the bazel
+    # call into yarn from also following the yarn-path as desired.
+    if not is_windows:
+        # Yarn entry point
+        repository_ctx.file(
+            "bin/yarn",
+            content = """#!/usr/bin/env bash
+# Generated by node_repositories.bzl
+# Immediately exit if any command fails.
+set -e
+unset YARN_IGNORE_PATH
+{get_script_dir}
+"{node}" "$SCRIPT_DIR/{script}" "$@"
+""".format(
+                get_script_dir = GET_SCRIPT_DIR,
+                node = repository_ctx.path(node_label),
+                script = yarn_script_relative,
+            ),
+            executable = True,
+        )
+    else:
+        # Yarn entry point
+        repository_ctx.file(
+            "bin/yarn.cmd",
+            content = """@echo off
+SET SCRIPT_DIR=%~dp0
+SET "YARN_IGNORE_PATH="
+"{node}" "%SCRIPT_DIR%\\{script}" %*
+""".format(
+                node = repository_ctx.path(node_label),
+                script = yarn_script_relative,
+            ),
+            executable = True,
+        )
+
+    # Base BUILD file for this repository
+    build_content = """# Generated by node_repositories.bzl
+package(default_visibility = ["//visibility:public"])
+exports_files(["{yarn_entry}"])
+alias(name = "yarn_bin", actual = "{yarn_bin_label}")
+alias(name = "yarn", actual = "{yarn_entry}")
+filegroup(
+  name = "yarn_files",
+  srcs = {yarn_files_glob},
+)
+""".format(
+        yarn_bin_export = ("\n  \"%s\"," % yarn_bin),
+        yarn_files_glob = "glob([\"bin/yarnpkg/**\"])",
+        yarn_bin_label = yarn_bin_label,
+        yarn_entry = yarn_entry,
+    )
+
+    repository_ctx.file("BUILD.bazel", content = build_content)
+
+yarn_repositories = repository_rule(
+    implementation = _yarn_repositories_impl,
+    attrs = {
+        "node_repository": attr.string(
+            default = "nodejs",
+            doc = """The basename for a nodejs toolchain to use for running yarn.
+            Usually this is the value of the `name` attribute given to a nodejs_register_toolchains call in WORKSPACE""",
+        ),
+        "yarn_download_auth": attr.string_dict(
+            default = {},
+            doc = """auth to use for all url requests
+    Example: {\"type\": \"basic\", \"login\": \"<UserName>\", \"password\": \"<Password>\" }
+    """,
+        ),
```

Comment:
Was this here before? Seems kinda insecure

Reply:
yes, this block was moved without changes from the original to avoid making it more breaking than necessary